src/server/active_tran_server.cpp

```diff
@@ -173,9 +190,14 @@ void active_tran_server::receive_log_page (cubpacking::unpacker &upk)
 
   if (error_code == NO_ERROR)
     {
-      char log_page_buffer[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT];
-      LOG_PAGE *log_page = reinterpret_cast<LOG_PAGE *> (PTR_ALIGN (log_page_buffer, MAX_ALIGNMENT));
+
+      char *log_page_buffer = new char [db_log_page_size ()];
+      LOG_PAGE *log_page = reinterpret_cast<LOG_PAGE *> (log_page_buffer);
       std::memcpy (log_page, message.c_str () + sizeof (error_code), db_log_page_size ());
+
+      std::shared_ptr<LOG_PAGE> shared_log_page (log_page);
+      m_async_log_page_receiver->set_page (std::move (shared_log_page));
```

Comment:
Setting char[] to std::shared_ptr<LOG_PAGE> may not be safe. Please find a way to make sure memory allocator/deallocator match.